modules/porous_flow/include/utils/PorousFlowVanGenuchten.h

```diff
@@ -137,4 +137,135 @@ Real dRelativePermeabilityNW(Real seff, Real m);
  * @return second derivative of relative permeability wrt effective saturation
  */
 Real d2RelativePermeabilityNW(Real seff, Real m);
+
+/**
+ * Parameters associated with the extension of the hysteretic capillary pressure function to low
+ * saturation values
+ * @ ExtensionStrategy the type of extension used
+ * @ S liquid saturation at the point of extension
+ * @ Pc capillary pressure at the point of extension
+ * @ dPc d(Pc)/dS at the point of extension
+ */
+struct LowCapillaryPressureExtension
+{
+  enum ExtensionStrategy
+  {
+    NONE,
+    QUADRATIC,
+    EXPONENTIAL
+  } strategy;
+  Real S;
+  Real Pc;
+  Real dPc;
```

Comment:
For clarity please separate the definition of the enum from the member variable.
Can/should these be const members?

```suggestion
  enum ExtensionStrategy
  {
    NONE,
    QUADRATIC,
    EXPONENTIAL
  };
  const ExtensionStrategy strategy;
  const Real S;
  const Real Pc;
  const Real dPc;
```

Reply:
Separation is done.  I *think* the members can be const - i'll try

Reply:
I don't know that they need to be. It was more of a question. Do you envision a use where you would create one of these structs and need to change values. If that doesn't make sense then make them const, if it does leave it alone.

Reply:
I like the idea of const, @aeslaughter , thanks.   In the current code i think it makes sense.  I'm just re-writing stuff, as with const i need some constructors.  Later on if const becomes annoying i can always remove it.

Reply:
done